### PR TITLE
Close #6 - Load command params from a config file

### LINKED ISSUE #6: Introduce config file support (cosmiconfig package)
```
{
  ast: 'path/to/ast',
  plugin: 'concordialang-ui-html',
  outputDir: 'path/to/output/dir'
}
```

### PR DESCRIPTION
**Pull request by WillianGoncalves**

### CODE REVIEW

File: src/commands/generate.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import {Command, flags} from '@oclif/command'
 import {AstProcessor, ProcessResult} from 'concordialang-ui-core'
+const cosmiconfig = require('cosmiconfig')
```
[the remark this answers is not in the document]
Comment (WillianGoncalves) [reply src/commands/generate.ts]: No, the lint shows an error.

[the remark this answers is not in the document]
Comment (WillianGoncalves) [reply src/commands/generate.ts]: No. I tried but I got the message: "Could not find a declaration file for module 'cosmiconfig'".

File: src/commands/generate.ts [hunk 2]
```diff
@@ -12,25 +13,47 @@ export default class Generate extends Command {
...
+    const {flags} = this.parse(Generate)
+
+    const ast: string = this.getParamFromConfigFile('ast') || flags.ast
+    const pluginName: string = this.getParamFromConfigFile('plugin') || flags.plugin
+    const outputDir: string = this.getParamFromConfigFile('outputDir') || flags.outputDir
+
+    return {ast, pluginName, outputDir}
+  }
+
+  private getParamFromConfigFile(param: string): any {
```
Comment (WillianGoncalves) [inline src/commands/generate.ts]: There is a better implementation for this function.

File: src/commands/generate.ts [hunk 3]
```diff
@@ -12,25 +13,47 @@ export default class Generate extends Command {
...
+      this.error(`Plugin ${pluginName} not found`)
     }
   }
+
+  private getCommandParams(): any {
+    const {flags} = this.parse(Generate)
+
+    const ast: string = this.getParamFromConfigFile('ast') || flags.ast
+    const pluginName: string = this.getParamFromConfigFile('plugin') || flags.plugin
+    const outputDir: string = this.getParamFromConfigFile('outputDir') || flags.outputDir
```
Comment (thiagodp) [inline src/commands/generate.ts]: I suggest
```javascript
const outputDir: string = flags.outputDir
    || this.getParamFromConfigFile('outputDir')
    || DEFAULT_OUTPUT_DIR
```
where `DEFAULT_OUTPUT_DIR`'s value could be `.`

**Comment by vcpablo** [comment]
I believe that the correct order of priority for the command options would be
1. Flags
2. Config file
3. Default value

Another thing is, if no `outputDir` is defined neither through flag or config file, the prototypes should be generated in the current directory.

File: src/commands/generate.ts [hunk 4]
```diff
@@ -12,25 +13,47 @@ export default class Generate extends Command {
...
+      await command.run(['--features', JSON.stringify(result), '--outputDir', outputDir])
+    } else {
+      this.error(`Plugin ${pluginName} not found`)
     }
   }
+
+  private getCommandParams(): any {
+    const {flags} = this.parse(Generate)
+
+    const ast: string = this.getParamFromConfigFile('ast') || flags.ast
```
Comment (thiagodp) [inline src/commands/generate.ts]: Flag first

File: src/commands/generate.ts [hunk 5]
```diff
@@ -12,25 +13,47 @@ export default class Generate extends Command {
...
+    const pluginName: string = this.getParamFromConfigFile('plugin') || flags.plugin
```
Comment (thiagodp) [inline src/commands/generate.ts]: Flag first

File: src/commands/generate.ts [hunk 6]
```diff
@@ -12,25 +13,47 @@ export default class Generate extends Command {
 
   static flags = {
```
Comment (thiagodp) [inline src/commands/generate.ts]: Better representing the object's structure with an interface

File: src/commands/generate.ts [hunk 7]
```diff
@@ -12,25 +13,47 @@ export default class Generate extends Command {
...
+
+    if (plugin) {
+      const command = plugin.findCommand('generate')
+      await command.run(['--features', JSON.stringify(result), '--outputDir', outputDir])
+    } else {
+      this.error(`Plugin ${pluginName} not found`)
     }
   }
+
+  private getCommandParams(): any {
```
Comment (thiagodp) [inline src/commands/generate.ts]: You can use the mentioned interface here, as a return type

File: src/commands/generate.ts [hunk 8]
```diff
@@ -12,25 +13,47 @@ export default class Generate extends Command {
...
+    const {flags} = this.parse(Generate)
+
+    const ast: string = this.getParamFromConfigFile('ast') || flags.ast
+    const pluginName: string = this.getParamFromConfigFile('plugin') || flags.plugin
+    const outputDir: string = this.getParamFromConfigFile('outputDir') || flags.outputDir
+
+    return {ast, pluginName, outputDir}
```
Comment (thiagodp) [inline src/commands/generate.ts]: Why not using `plugin` instead of `pluginName` ?

**Pull request merged**